worker/upload.go

```diff
@@ -292,10 +293,17 @@ func (mgr *uploadManager) Upload(ctx context.Context, r io.Reader, rs api.Redund
 	defer close(nextSlabChan)
 
 	// create the response channel
-	var wg sync.WaitGroup
+	var ongoingUploads uint64
 	respChan := make(chan slabUploadResponse)
 	defer func() {
-		wg.Wait()
+		// wait for ongoing uploads to send their response down the channel before closing it
+		if atomic.LoadUint64(&ongoingUploads) > 0 {
```

Comment:
I think there is a race here. Theoretically this `if` can trigger right before the counter is decremented. Then we are in a spot where no response will be sent over the channel anymore to trigger the `for` loop.

Reply:
hm ok I'll revert to my original idea of not necessarily `return`'ing on the first error we encounter but instead keep it around and return when ongoingUploads is 0